src/Microsoft.ML.Data/Transforms/TypeConverting.cs

```diff
@@ -538,33 +514,63 @@ private bool SaveAsOnnxCore(OnnxContext ctx, int iinfo, string srcVariableName,
+            public readonly DataKind OutputKind;
+            public readonly KeyCount OutputKeyCount;
+
+            /// <summary>
+            /// Describes how the transformer handles one column pair.
+            /// </summary>
+            /// <param name="name">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
+            /// <param name="outputKind">The expected kind of the converted column.</param>
+            /// <param name="inputColumnName">Name of column to transform. If set to <see langword="null"/>, the value of the <paramref name="name"/> will be used as source.</param>
+            /// <param name="outputKeyCount">New key range, if we work with key type.</param>
+            public ColumnInfo(string name, DataKind outputKind, string inputColumnName, KeyCount outputKeyCount = null)
+            {
+                Name = name;
+                InputColumnName = inputColumnName ?? name;
+                OutputKind = outputKind;
+                OutputKeyCount = outputKeyCount;
+            }
+        }
+
         /// <summary>
         /// Convinence constructor for simple one column case.
         /// </summary>
         /// <param name="env">Host Environment.</param>
         /// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
         /// <param name="inputColumnName">Name of the column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
         /// <param name="outputKind">The expected type of the converted column.</param>
-        public TypeConvertingEstimator(IHostEnvironment env,
+        internal TypeConvertingEstimator(IHostEnvironment env,
             string outputColumnName, string inputColumnName = null,
             DataKind outputKind = Defaults.DefaultOutputKind)
-            : this(env, new TypeConvertingTransformer.ColumnInfo(outputColumnName, outputKind, inputColumnName ?? outputColumnName))
+            : this(env, new ColumnInfo(outputColumnName, outputKind, inputColumnName ?? outputColumnName))
         {
         }
 
         /// <summary>
         /// Create a <see cref="TypeConvertingEstimator"/> that takes multiple pairs of columns.
         /// </summary>
-        public TypeConvertingEstimator(IHostEnvironment env, params TypeConvertingTransformer.ColumnInfo[] columns) :
+        internal TypeConvertingEstimator(IHostEnvironment env, params ColumnInfo[] columns) :
             base(Contracts.CheckRef(env, nameof(env)).Register(nameof(TypeConvertingEstimator)), new TypeConvertingTransformer(env, columns))
         {
         }
 
+        /// <summary>
+        /// Returns the <see cref="SchemaShape"/> of the schema which will be produced by the transformer.
+        /// Used for schema propagation and verification in a pipeline.
+        /// </summary>
```

Comment:
Please disregard my comment, apparently it doesn't work for intellisense. [https://github.com/dotnet/csharplang/issues/313](https://github.com/dotnet/csharplang/issues/313)

---
In reply to: [253150851](https://github.com/dotnet/machinelearning/pull/2365#discussion_r253150851) [](ancestors = 253150851)

Reply:
Yes we were talking about this with senja yesterday! it would be super useful!

---
In reply to: [253154622](https://github.com/dotnet/machinelearning/pull/2365#discussion_r253154622) [](ancestors = 253154622,253150851)